jupyter_server/base/handlers.py

```diff
@@ -589,7 +589,7 @@ def check_host(self) -> bool:
             )
         return allow
 
-    async def prepare(self) -> Awaitable[None] | None:  # type:ignore[override]
+    async def prepare(self, *, _redirect_to_login=True) -> Awaitable[None] | None:  # type:ignore[override]
```

Comment:
private keyword arg makes this hard for subclasses, e.g. extensions with websockets, but I suppose they will inherit from our own websocket classes, too?

I wonder if this should be a Handler class attribute, rather than a private argument.

Reply:
There is a check for inheritance from `JupyterHandler` in the weboscket mixin class:

https://github.com/jupyter-server/jupyter_server/blob/c3f5d8fc4bd78aaf67b34e7f1e1aaa619d578899/jupyter_server/base/websocket.py#L110-L121

A Handler class attribute is certainly a reasonable alternative that I briefly considered. My thinking was along: if you pass `_redirect_to_login` argument but are inheriting from wrong class you will get an error. If you set a class attribute but are inheriting from a wrong class you would not get anything. The downside of passing argument is that mypy typing might complain downstream.